blog/_posts/2024-08-19-KAITO-finetuning.md

```diff
@@ -0,0 +1,131 @@
+  adapters:
+    - source:
+        name: myadapter
+        image: myregistry.azurecr.io/adapters/myadapter:0.0.1
+        imagePullSecrets:
+          - myregistrysecret
+      strength: "1.0"
+```
+
+As new inference requests come in, they’ll flow through the model that merges the adapters and the response will be affected by the newly trained data. 
+Great! After updating our inference endpoint to leverage the new adapter, let’s ask the same question and check the response from the fine-tuned phi-3-mini-128k-instruct model:
+
+
+![Screenshot of tuned model response](/blog/assets/images/kaito_tuned_phi3_response.png)
+*Screenshot of tuned model response*
+
+That definitely looks more accurate – the fine-tuned model quickly discerned that we’re looking for an answer related to Kubernetes and provided a much better response. However, there’s always room to improve the model’s accuracy for domain specific questions. We can improve upon or find new input datasets to create new tuning adapters. This iterative tuning process can be done by repeating the steps described above.
+
+## Other options to make models smarter
+
+Retrieval-Augmented Generation (RAG) is another common technique used to improve the inference accuracy of foundation models. Compared to LoRA fine-tuning, RAG eliminates the need of training Jobs to generate adapters. It has a more complicated workflow with additional components like a vector database, indexing and query servers, etc. The good news is that RAG support with a simple user experience is in KAITO’s roadmap. 🚀
+
+## Summary
+
+Users expect LLMs to stay on task and provide accurate answers. While foundation models on their own have a wealth of knowledge, they often need to be refined for specific tasks. This can be achieved by leveraging techniques like fine-tuning or RAG, but the process of implementing these techniques is not always straightforward. As you saw, with KAITO, it becomes **really easy**. All you need to do is specify an input dataset, a GPU SKU, and off it goes! KAITO will take care of the rest. Its adapter design makes it very portable to attach to multiple workspaces or even attach multiple adapters to a single workspace. In the end, we can see that the inference and tuning workspaces along with adapters can become fundamental building blocks in your overall ML workflow.
+
+We’re just getting started and would love your feedback. To learn more about KAITO fine-tuning, AI model deployment on AKS or KAITO’s roadmap in general, check out the following links:
+
+**Resources**
+
+- [KAITO fine-tuning guide](https://github.com/Azure/kaito/tree/main/docs/tuning)
+- [Concepts - Fine-tuning language models for AI and machine learning workflows](https://learn.microsoft.com/azure/aks/concepts-fine-tune-language-models)
+- [Concepts - Small and large language models](https://learn.microsoft.com/azure/aks/concepts-ai-ml-language-models)
+- [KAITO Roadmap](https://github.com/orgs/Azure/projects/669)
```

Comment:
```suggestion
- [KAITO Roadmap](https://github.com/orgs/Azure/projects/669)

## Acknowledgements
Huge thanks to [Paul Yu](https://www.linkedin.com/in/yupaul) for co-authoring this blog and providing valuable feedback on KAITO since the early stages of the project!
```